bananalyzer/runner/runner.py

```diff
@@ -22,25 +23,42 @@ class BananalyzerTest(BaseModel):
 class TestGenerator:
     def __init__(self) -> None:
         self._names: Dict[str, int] = {}
+        self._classnames: Dict[str, int] = {}
 
-    def generate_test(self, example: Example, headless: bool) -> BananalyzerTest:
+    def generate_test(self, example: Example) -> BananalyzerTest:
         return BananalyzerTest(
             code=f"""
 @pytest.mark.asyncio
-async def {self._generate_name(example)}() -> None:
-    async with async_playwright() as p:
-        browser = await p.chromium.launch(headless={headless})
-        context = await browser.new_context()
-        example = get_example_by_url("{example.url}")
+class {self._generate_class_name(example)}:
```

Comment:
Tests are now group as one Class per example. This allows us to run the agent once and then then run a series of tests against the result